internal/resources/statefulsets.go

```diff
@@ -27,11 +27,13 @@ func buildStatefulSet(
 	planner *pln.Planner,
 	dataPVCName, statePVCName, ns string) *appsv1.StatefulSet {
 	// ---
-	labels := labelsForSmbServer(planner.InstanceName())
+	labels := labelsForSmbServer(planner)
 	size := planner.ClusterSize()
 	podSpec := buildClusteredPodSpec(planner, dataPVCName, statePVCName)
 	if planner.NodeSpread() {
-		podSpec.Affinity = buildOneSmbdPerNodeAffinity(labels, serviceLabel)
+		podSpec.Affinity = buildOneSmbdPerNodeAffinity(planner, labels, serviceLabel)
+	} else {
+		podSpec.Affinity = affinityForSmbPod(planner)
```

Comment:
In a case with no specification to _PodSettings.Affinity_ in _SmbCommonConfig_ does the logic still holds true when _affinityForSmbPod()_ returns _nil_ here? My intention is to ask whether a check is required in _affinityForSmbPod()_ for _SmbCommonConfig.PodSettings.Affinity_.

Reply:
If I understand the question, you want to know what happens if there *is* a PodSettings but Affinity is not set in the YAML? Because, if there's any value given for Affinity the object will not be set to null.  In that case it should be the same as if PodSettings was not given (is null) the affinityForSmbPod function returns null.

Does that make sense? If not could you restate the q, thanks!

Reply:
Also the "zero value" for podSpec.Affinity is null, so if affinityForSmbPod returns null, assigning null to `podSpec.Affinity` is rougly a no-op IIUC.

Reply:
> If I understand the question, you want to know what happens if there _is_ a PodSettings but Affinity is not set in the YAML? Because, if there's any value given for Affinity the object will not be set to null. In that case it should be the same as if PodSettings was not given (is null) the affinityForSmbPod function returns null.
> 
> Does that make sense? If not could you restate the q, thanks!

You got it right. Thanks for the clarification.